### PR TITLE
lightning: fix checksum mismatch when parallel import data

### LINKED ISSUE #29692: lightning checksum error when parallel import data
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. Use 2 lightning instance to parallel import data to TiDB, no data conflict in the data source

### 2. What did you expect to see? (Required)
1. Lightning import should succeed 

### 3. What did you see instead (Required)
1. Lighting reports checksum error 
```
Error: checksum mismatched remote vs local => (checksum: 11222303442876741793 vs 3225814776482587640) (total_kvs: 18161516 vs 9080796) (total_bytes:3368780389 vs 1684395306)\ntidb lightning encountered error:  checksum mismatched remote vs local => (checksum: 11222303442876741793 vs 3225814776482587640) (total_kvs: 18161516 vs 9080796) (total_bytes:3368780389 vs 1684395306)
```

### 4. What is your TiDB version? (Required)
/ # /tidb-lightning -V
Release Version: v5.3.0
Git Commit Hash: c6d65239a01662b239ff6f6c10abda4ab28ec8c5
Git Branch: heads/refs/tags/v5.3.0
Go Version: go1.16.4
UTC Build Time: 2021-11-10 14:29:25
Race Enabled: false

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Issue Number: close #29692

Problem Summary:

lightning checksum error when parallel import data.

### What is changed and how it works?

Don't call FinishTable when other lightning will calculate checksum.

### Check List

Tests 

- [x] Integration test

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: br/tests/lightning_distributed_import/run.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+run_lightning --backend local --sorted-kv-dir "$TEST_DIR/lightning_distributed_import.sorted1" \
+  -d "tests/$TEST_NAME/data1" --log-file "$LOG_FILE1" --config "tests/$TEST_NAME/config.toml" &
+pid1="$!"
+
+run_lightning --backend local --sorted-kv-dir "$TEST_DIR/lightning_distributed_import.sorted2" \
+  -d "tests/$TEST_NAME/data2" --log-file "$LOG_FILE2" --config "tests/$TEST_NAME/config.toml" &
+pid2="$!"
+
+wait "$pid1" "$pid2"
```
Comment (kennytm) [inline br/tests/lightning_distributed_import/run.sh:33]: should at least add a `select count(*) from distributed_import.t` and check that equals 10 afterwards

**Review by kennytm — COMMENTED** [review]
rest LGTM

File: br/tests/lightning_distributed_import/run.sh (hunk 1 above)
Comment (gozssky) [reply br/tests/lightning_distributed_import/run.sh:33]: done

File: br/pkg/lightning/restore/table_restore.go [hunk 2]
```diff
@@ -796,7 +796,8 @@ func (tr *TableRestore) postProcess(
 			nextStage = checkpoints.CheckpointStatusChecksumSkipped
 		}
 
-		if err == nil {
+		// Don't call FinishTable when other lightning will calculate checksum.
+		if err == nil && needChecksum {
```
Comment (glorv) [inline br/pkg/lightning/restore/table_restore.go:800]:
```suggestion
		if err == nil && !hasDupe && needChecksum {
```

**Review by glorv — COMMENTED** [review]
rest LGTM

**Review by kennytm — APPROVED** [review]

**Review by glorv — APPROVED** [review]

**Pull request merged**